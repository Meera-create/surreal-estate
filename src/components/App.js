
import '../styles/app.css';
import NavBar from './navbar';
import {Routes, Route} from 'react-router-dom'
import Properties from './properties';
import AddProperty from './addproperty';
import React from 'react';


const App =() => {
  return (
    <div className="App">
      
        <NavBar />
        <Routes>
        <Route path="/" element={<Properties />} />
    
        <Route path="/add-property" element={<AddProperty />} />
        
        </Routes>
          
    </div>
  );
}

export default App;
