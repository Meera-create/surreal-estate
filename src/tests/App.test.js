import { render } from '@testing-library/react';
import App from '../components/App';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

describe('testing app component', () => {
  test('checks if app renders', () => {
    const { asFragment } = render(<BrowserRouter><App /></BrowserRouter>)
    expect(asFragment()).toMatchSnapshot();
  })
});


