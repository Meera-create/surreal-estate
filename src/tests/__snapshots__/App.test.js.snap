// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing app component checks if app renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar_box"
    >
      <nav
        class="navbar"
      >
        <div>
          <img
            alt="logo_image"
            src="../styles/SURREAL ESTATE.jpg"
          />
          <ul
            class="navbar-links-item"
          >
            <li>
              <a
                class="item"
                href="/"
              >
                View Properties
              </a>
            </li>
            <li>
              <a
                class="item"
                href="/add-property"
              >
                Add a property
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div>
      <div
        class="title"
      >
        Properties Page
      </div>
      <div
        class="properties"
      >
        <div
          class="property_lists"
        >
          <h1
            class="title"
          >
            List of Properties
          </h1>
          <h2
            class="property_title"
          />
          <h3
            class="price"
          />
          <h3
            class="location"
          >
            ,
          </h3>
          <p
            class="property_details"
          >
            bathrooms:, bedrooms:
          </p>
          <p
            class="email"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
